adapters/zmaticoo/zmaticoo.go

```diff
@@ -0,0 +1,189 @@
+package zmaticoo
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"text/template"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	EndpointTemplate *template.Template
+}
+
+// Builder builds a new instance of the zmaticoo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		EndpointTemplate: template,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	adapterRequest, errs := a.makeRequest(request)
+	if errs == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	}
+
+	return adapterRequests, errs
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	zmaticooExt, errs := getZmaticooExt(request)
+
+	if zmaticooExt == nil {
+		return nil, errs
+	}
+	endPoint, err := a.getEndpoint(zmaticooExt)
+	if err != nil {
+		return nil, append(errs, err)
+	}
+	transform(request)
+	reqBody, err := json.Marshal(request)
+
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endPoint,
+		Body:    reqBody,
+		Headers: headers,
+	}, errs
+}
+
+func transform(request *openrtb2.BidRequest) {
+	for i, imp := range request.Imp {
+		if imp.Native != nil {
+			var nativeRequest map[string]interface{}
+			nativeCopyRequest := make(map[string]interface{})
+			err := json.Unmarshal([]byte(request.Imp[i].Native.Request), &nativeRequest)
+			//just ignore the bad native request
```

Comment:
you are ignoring the bad native request. I recommend to return error to this function. This will help you in debugging the issue for the publisher's request that you are not responding with bid.

Reply:
This was missed.

Reply:
i have resovled

Reply:
> i have resovled

@lxj15398019970  could you point out changes that have resolved the comment

Reply:
@lxj15398019970  requesting to address open comment